security/RFP.md

```diff
@@ -0,0 +1,254 @@
+
+### Artifact 2: Coordinated Hydra Head V1 Specification
+
+The Hydra Head protocol implementation derives from the original publication in several ways. Especially some simplification have been introduced and generalizations removed.
+
+The [Coordinated Hydra Head V1 specification](https://docs.google.com/document/d/1XQ0C7Ko3Ifo5a4TOcW1fDT8gMYryB54PCEgOiFaAwGE/) captures these deviations and also includes the "formal notation" of the actual transaction constraints (which are foregone in the original paper). Also, it details the L2 protocol logic for the **Coordinated** Head protocol - which is implemented in V1.
+
+Note that it is lacking some structure and introductory sections and we recommend to see Artifact 1 for that.
+
+FIXME the following list is probably not usefull since it should be in the spec
+In particular, the following simplifications are done in the actual implementation:
+* Kagg is a list of keys
+* snapshot is signed for every transaction and only snapshot can be used in close transaction
+* ...
+
+### Artifact 3: Hydra Head Protocol Implementation
+
+With Hydra Head Protocol Implementation we refer to the software component that is used to operate a node in the Hydra Head protocol. The `hydra-node` allows its users to open a head, lock funds in it, connect to peers, process transactions as a layer 2, close a head and unlock the corresponding funds. It includes Hydra plutus scripts, Hydra head chain layer, layer 2 code, network communication between peers, and an API for clients to connect and use the node.
+
+The implementation can be found in this [Github repository](https://github.com/input-output-hk/hydra-poc)
+
+Version to be audited: [0.9.0](https://github.com/input-output-hk/hydra-poc/releases/tag/0.9.0)
+
+TODO describe the inputs and outputs of a hydra node
+
+TODO: clarify which artifacts we need to introduce
+
+TODO:We should share our test architecture/topology and also the test results here: https://hydra.family/head-protocol/benchmarks/tests/hydra-cluster/hspec-results
+
+#### Artifact 3.3: Hydra plutus scripts
+
+TODO
+
+Grab stuff from https://hydra.family/head-protocol/haddock/hydra-plutus/index.html
+
+#### Artifact 3.4: Hydra node chain layer code
+
+TODO
+
+Grab stuff from https://hydra.family/head-protocol/haddock/hydra-node/index.html sub-sections of _Hydra.Chain.Direct_
+
+#### Artifact 3.5: layer 2 code
+
+TODO
+
+## Hypotheses
+
+The Coordinated Hydra Head V1 Specification defines several hypotheses under which the security properties defined hold. Any assessment performed during this audit must be done under these hypotheses.
```

Comment:
I don't think we should be using the same assumptions for the implementation. The specification defines a more narrow space to reason about some things. If we consider the implementation we cannot simply assume (for example) "all network connections are authenticated and never fail".